lib/foreman_tasks/engine.rb

```diff
@@ -16,118 +16,120 @@ class Engine < ::Rails::Engine
+                         'This can be used to prevent polling too frequently for long running tasks.'),
+                    default: 1,
+                    full_name: N_("Polling intervals multiplier"),
+                    validate: { numericality: { greater_than: 0 } })
           end
+        end
 
-          register_graphql_query_field :task, '::Types::Task', :record_field
-          register_graphql_query_field :tasks, '::Types::Task', :collection_field
-          register_graphql_query_field :recurring_logic, '::Types::RecurringLogic', :record_field
-          register_graphql_query_field :recurring_logics, '::Types::RecurringLogic', :collection_field
+        register_graphql_query_field :task, '::Types::Task', :record_field
+        register_graphql_query_field :tasks, '::Types::Task', :collection_field
+        register_graphql_query_field :recurring_logic, '::Types::RecurringLogic', :record_field
+        register_graphql_query_field :recurring_logics, '::Types::RecurringLogic', :collection_field
 
-          register_graphql_mutation_field :cancel_recurring_logic, '::Mutations::RecurringLogics::Cancel'
+        register_graphql_mutation_field :cancel_recurring_logic, '::Mutations::RecurringLogics::Cancel'
 
-          logger :dynflow, :enabled => true
-          logger :action, :enabled => true
+        logger :dynflow, :enabled => true
+        logger :action, :enabled => true
 
-          role 'Tasks Manager', [:view_foreman_tasks, :edit_foreman_tasks],
-               'Role granting permissions to inspect, cancel, resume and unlock tasks'
-          role 'Tasks Reader', [:view_foreman_tasks],
-               'Role granting permissions to inspect tasks'
+        role 'Tasks Manager', [:view_foreman_tasks, :edit_foreman_tasks],
+             'Role granting permissions to inspect, cancel, resume and unlock tasks'
+        role 'Tasks Reader', [:view_foreman_tasks],
+             'Role granting permissions to inspect tasks'
 
-          widget 'foreman_tasks/tasks/dashboard/tasks_status', :sizex => 6, :sizey => 1, :name => N_('Task Status')
-          widget 'foreman_tasks/tasks/dashboard/latest_tasks_in_error_warning', :sizex => 6, :sizey => 1, :name => N_('Latest Warning/Error Tasks')
+        widget 'foreman_tasks/tasks/dashboard/tasks_status', :sizex => 6, :sizey => 1, :name => N_('Task Status')
+        widget 'foreman_tasks/tasks/dashboard/latest_tasks_in_error_warning', :sizex => 6, :sizey => 1, :name => N_('Latest Warning/Error Tasks')
 
-          register_gettext domain: "foreman_tasks"
+        register_gettext domain: "foreman_tasks"
+      end
+    end
 
-          ForemanTasks.dynflow.eager_load_actions!
-          extend_observable_events(::Dynflow::Action.descendants.select { |klass| klass <= ::Actions::ObservableAction }.map(&:namespaced_event_names))
-        end
+    initializer 'foreman_tasks.observable_events' do |app|
```

Comment:
This looks familiar :) https://github.com/theforeman/foreman-tasks/pull/752#discussion_r1600208455

Although, I don't remember why I dropped that idea, but it still seems logical.

Today, we put the whole registration block under `to_prepare` in probably all the plugins just because it made it work with Zeitwerk. Although, it was never suggested as the right thing to do, just the one that works. We might want to revisit that again...

Reply:
It seems I'm retracing your footsteps! I found another way to do it using after_initialize instead of to_prepare. I think it is closer to the original intent, yet allowing calls to application code. Pushed a new commit so you can take a look